### PR TITLE
[MRG] Remove boston dataset following sklearn deprecation, replace by leukemia

### PR DESCRIPTION
**Pull request by mathurinm**

### CODE REVIEW

**Review by tomMoral — COMMENTED** [review]
LGTM! thx @mathurinm 

what's the size of leukemia? did you run the benchmark on it? the nice thing about Boston was it was super quick, what about the new one?

**Review by agramfort — APPROVED** [review]

**Comment by mathurinm** [comment]
`benchopt run . -d libsvm\[dataset=bodyfat\]` runs in seconds now

**Review by agramfort — COMMENTED** [review]
I would not create 2 files for this. Maybe we just one dataset called libsvm? or you think it makes things hard to find? note that you will have libsvm['bodyfat'] in all figures  / legends etc.

File: datasets/libsvm.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+with safe_import_context() as import_ctx:
+    from libsvmdata import fetch_libsvm
+
+
+class Dataset(BaseDataset):
+
+    name = "libsvm"
+
+    parameters = {
+        'dataset': ["bodyfat"],
```
Comment (tomMoral) [inline datasets/libsvm.py:15]:
```suggestion
        'dataset': ["bodyfat", "leukemia"],
```

**Review by tomMoral — COMMENTED** [review]
leukemia

**Comment by mathurinm** [comment]
Done. beware that leukemia from libsvm is smaller than leukemia from open_ml (it corresponds to the 38 training samples)

**Review by agramfort — APPROVED** [review]

**Comment by agramfort** [comment]
Arfff the leukemia you find in papers is 72 samples. I think it’s misleading as this is now. Maybe you fetch train and test and you concat them? Just thinking out loud but with no strong conviction

**Comment by mathurinm** [comment]
That is exactly why I originally went with a separate leukemia file, *fetched from open ml*, which is the one with 72 samples. 

If the user wants to use leukemia from LIBSVM, then IMO he should expect it to respect the doc on LIBSVM website, meaning to have 38 samples (or 34 for test).

**Comment by agramfort** [comment]
Then I think we agree we should revert to 2 files …. Sorry for the mess

**Comment by mathurinm** [comment]
No problem :) 
Good to go on my side

**Review by agramfort — APPROVED** [review]

File: datasets/leukemia.py [hunk 2]
```diff
@@ -4,20 +4,22 @@
 
 
 with safe_import_context() as import_ctx:
-    from sklearn.datasets import load_boston
+    from sklearn.datasets import fetch_openml
+    from sklearn.preprocessing import LabelBinarizer
 
 
 class Dataset(BaseDataset):
```
Comment (tomMoral) [inline datasets/leukemia.py:11]: I would add a comment stating why we have 2 different datasets.

**Review by tomMoral — APPROVED** [review]
LGTM up to adding a comment to explain why we have 2 similar datasets.

File: datasets/leukemia.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+class Dataset(BaseDataset):
+
+    name = "leukemia"
+
+    install_cmd = 'conda'
+    requirements = ['scikit-learn']
+
+    def get_data(self):
+        # this is the whole leukemia with train + test data (72 samples)
```
Comment (tomMoral) [inline datasets/leukemia.py:19]:
```suggestion
        # Unlike libsvm[leukemia], this dataset corresponds to the whole
        # leukemia  data with train + test data (72 samples) and not just
        # the training set.
```

**Review by agramfort — APPROVED** [review]

**Pull request merged**